app/src/org/commcare/android/resource/analytics/InstallAttempts.java

```diff
@@ -0,0 +1,69 @@
+package org.commcare.android.resource.analytics;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Stores install attempt data for a given resource in the update table.
+ *
+ * @param <A> Type of data stored at each install attempt. Must be
+ *            instantiated with a class that implements Serializeable.
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+class InstallAttempts<A> implements Serializable {
```

Comment:
Left this generic in case we want to keep track of different types of information. This is a result of not spec'ing out this part of the feature very well. Should get narrowed soon, when I revisit this part of the feater